luarules/gadgets/unit_air_attacksafetydistance.lua

```diff
@@ -13,20 +13,23 @@ function gadget:GetInfo()
 end
 
 --[[
-	Wiki: "attackSafetyDistance" movetypedata: Fighters abort dive toward target if within attackSafetyDistance and
+	springrts Wiki: "attackSafetyDistance" movetypedata: Fighters abort dive toward target if within attackSafetyDistance and
 	try to climb back to normal altitude while still moving toward target.	It's disabled by default. Set to half of
 	the minimum weapon range to avoid collisions, enemy fire, AOE damage. If set to greater than the weapon range,
 	the unit will fly over the target like a bomber.
 ]]--
 
 if not gadgetHandler:IsSyncedCode() then
-	return
+	return false
 end
 
 local isFighter = {}
 for udid, ud in pairs(UnitDefs) do
-	if ud.canFly and ud.customParams.fighter then
-		isFighter[udid] = true
+	if ud.canFly and ud.isStrafingAirUnit and ud.customParams.fighter then
+		-- customParams.fighter was orginally only checked if there, so contents didn't matter, instances were defined as 1 instead of true
+		local attackDistance = tonumber(ud.customParams.fighter) or 1
+		attackDistance = attackDistance == 1 and 300 or attackDistance
```

Comment:
Why the 1 -> 300 conversion? If there's existing units with fighter=1 then just set them to 300 and drop this code.